make.sh

```diff
@@ -63,8 +63,9 @@ rule() {
       ocb -no-plugin $TARGET.native &&
       cp _build/$TARGET.native goblint
     ;; debug)
-      ocb -tag debug $TARGET.d.byte &&
-      cp _build/$TARGET.d.byte goblint.byte
+      eval $(opam config env)
+      dune build $TARGET.bc &&
+      cp _build/default/$TARGET.bc goblint.byte
```

Comment:
Couldn't the dune target be named `goblint.byte` as well to be clear that it's the same thing?

Reply:
The target in `./src/dune` is `goblint.bc` because `.bc` is just the extension dune expects for byte-code files. Instead of the target in `./src/dune` it now uses the one in `./dune`.

Reply:
Ahh, ok, I wasn't aware that dune required the `.bc` extension itself. Then it will have to stay.

Although I think something is still off with the copying:
1. `dune` copies `_build/default/src/goblint.bc` → `_build/default/goblint.byte`
2. `make` now copies `_build/default/goblint.bc` → `goblint.byte`.

The intermediate doesn't match up it seems.

Reply:
>`make` now copies `_build/default/goblint.bc` → `goblint.byte`

Actually, it copied `_build/default/src/goblint.bc` -> `goblint.byte`, as `$TARGET` expands to `src/goblint`. 

Should I just get rid of the intermediate step in `./dune` and call the target `$TARGET$.bc` in `make.sh` (as it was at first)?

Reply:
It's a bit redundant then, but we also do it for normal compilation in that case. I guess what it ensures is that in `_build/default/` you also have a copy as if it's the top level of the repository/build directory.

It might even be necessary to keep it if one wants to `opam install` or `dune @install` since those would only have the build directory somewhere.

Reply:
Now it does the following:

1. `dune` copies `_build/default/src/goblint.bc` → `_build/default/goblint.byte`
2. `make` now copies `_build/default/goblint.byte` → `goblint.byte`.